transport/http/outbound.go

```diff
@@ -288,18 +288,22 @@ func (o *Outbound) callWithPeer(
 
 	span.SetTag("http.status_code", response.StatusCode)
 
+	bothResponseError := fromAcceptValue(response.Header.Get(BothResponseErrorHeader))
+	tres := &transport.Response{
+		Headers:          applicationHeaders.FromHTTPHeaders(response.Header, transport.NewHeaders()),
+		Body:             response.Body,
+		ApplicationError: fromApplicationStatusValue(response.Header.Get(ApplicationStatusHeader)),
+	}
+	if bothResponseError {
+		if response.StatusCode >= 300 {
+			return tres, getYARPCErrorFromResponse(response, true)
+		}
+		return tres, nil
+	}
 	if response.StatusCode >= 200 && response.StatusCode < 300 {
```

Comment:
we can put this check first right?  Once we've checked the status code the only other logic is:

if bothResponseError {
  return tres, get...(...)
}
return nil, get...(...)
```

Reply:
Can you explain more?

Reply:
Since Will is out, I think we’ll have to figure out what he’s suggesting.

I think he means that if we eliminate 1XX and 2XX first, we may be able to avoid the check for 300+ since it’s implied.

Reply:
I think this is the most readable solution, but that's just my opinion.